### PR TITLE
Showon doesn't work with new items in a subform field

### PR DESCRIPTION
**Pull request by tassosm**
Pull Request for Issue #29690

Make sure to run **npm i** before testing to rebuild JS files 

Probably resolves also #27176

### CODE REVIEW

**Comment by richard67** [comment]
I have tested this item :white_check_mark: successfully on b46dcec3308cbec3840158b2354d27c2efcd1f76<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/29691">issues.joomla.org/tracker/joomla-cms/29691</a>.</sub>

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on b46dcec3308cbec3840158b2354d27c2efcd1f76<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/29691">issues.joomla.org/tracker/joomla-cms/29691</a>.</sub>

**Comment by Quy** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/29691">issues.joomla.org/tracker/joomla-cms/29691</a>.</sub>

File: build/media_source/system/js/showon.es6.js [hunk 1]
```diff
@@ -220,10 +220,10 @@
     // Check is it subform, then wee need to fix some "showon" config
     if (target.classList.contains('subform-repeatable-group')) {
       const elements = [].slice.call(target.querySelectorAll('[data-showon]'));
-      const baseName = target.getAttribute('data-baseName');
+      const baseName = target.getAttribute('data-base-name');
```
Comment (dgrammatiko) [inline build/media_source/system/js/showon.es6.js]: My bad I guess. So basically I wanted to write: 
```js
const baseName = target.dataset.baseName;
```
and messed it up. Anyways using `dataset` here makes sense (there might be other instances in this file that need the same fix)

Comment (tassosm) [reply build/media_source/system/js/showon.es6.js]: It's now using `dataset`.

**Comment by richard67** [comment]
Back to pending.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/29691">issues.joomla.org/tracker/joomla-cms/29691</a>.</sub>

**Comment by richard67** [comment]
I have tested this item :white_check_mark: successfully on 57e43e6c59f528111a2717f3136f9ecf68684826<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/29691">issues.joomla.org/tracker/joomla-cms/29691</a>.</sub>

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on 57e43e6c59f528111a2717f3136f9ecf68684826<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/29691">issues.joomla.org/tracker/joomla-cms/29691</a>.</sub>

**Comment by Quy** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/29691">issues.joomla.org/tracker/joomla-cms/29691</a>.</sub>

**Comment by wilsonge** [comment]
Thanks!

**Pull request merged**